dashboard/app/models/user_level.rb

```diff
@@ -82,8 +92,35 @@ def paired?
   end
 
   def handle_unsubmit
-    if submitted_changed? from: true, to: false
+    if submitted_changed? from: true, to: false && !self.readonly_answers
```

Comment:
This case should actually never happen, as submitted being/becoming false and readonly_answers being true is impossible (which is now validated/enforced). I think I probably had a case where I was hitting this condition before this was enforced. I'll remove the `&& !self.readonly_answers` bit from this, and make sure nothing breaks.